src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect } from 'react';
+import { Icon, iconTypes } from '../Icon';
+import { SpanStyled } from './DatePicker.styles';
+import {
+    DivWrapperStyled,
+    InputStyled,
+    LabelStyled,
+    StrongStyled,
+} from '../Input/Input.styles';
+import color from '../../styles/colors';
+import { DatePickerProps } from './types';
+
+const DatePicker: React.FC<DatePickerProps> = ({
+    disabled = false,
+    id,
+    label,
+    name = 'Select a date',
+    onChange,
+    state = disabled ? 'disabled' : undefined,
+    type = 'date',
+    validation,
+    value = new Date().toISOString().substring(0, type === 'date' ? 10 : 7),
+}) => {
+    const [current, setCurrent] = useState(value);
+    const [currentState, setCurrentState] = useState(state);
+    const [invalidMessage, setInvalidMessage] = useState('');
```

Comment:
what about types for current state so we not have to call `setCurrentState(undefined)` ?